pkg/services/ngalert/api/api_ruler.go

```diff
@@ -259,6 +260,99 @@ func (srv RulerSrv) RouteGetRulesGroupConfig(c *contextmodel.ReqContext, namespa
 	return response.JSON(http.StatusAccepted, result)
 }
 
+// return list of rule groups paginated
+func (srv RulerSrv) RouteGetAlertRuleGroups(c *contextmodel.ReqContext) response.Response {
+	namespaceMap, err := srv.store.GetUserVisibleNamespaces(c.Req.Context(), c.SignedInUser.GetOrgID(), c.SignedInUser)
+	if err != nil {
+		return ErrResp(http.StatusInternalServerError, err, "failed to get namespaces visible to the user")
+	}
+
+	result := apimodels.NamespaceListResponse{
+		RuleGroups: make([]apimodels.RuleGroupSummary, 0),
+	}
+
+	if len(namespaceMap) == 0 {
+		srv.log.Debug("User has no access to any namespaces")
+		return response.JSON(http.StatusOK, result)
+	}
+
+	namespaceUIDs := make([]string, len(namespaceMap))
+	for k := range namespaceMap {
+		namespaceUIDs = append(namespaceUIDs, k)
+	}
+
+	const defaultPageSize = 100
```

Comment:
Not sure if this is just for draft, but let's not set a default page size and return them all if it's not specified.